### ISSUE TITLE
LS features fail due to unexpected character in package version

### ISSUE THREAD

**Issue by gayalkuruppu** [ballerina-platform/ballerina-lang#36470]
**Description:**
<img width="1640" alt="Screenshot 2022-06-05 at 21 20 34" src="https://user-images.githubusercontent.com/46120162/172523365-f277a212-64f0-4ef3-9d91-63563d9b0026.png">

**Steps to reproduce:**

**Affected Versions:**
2201.1.0

**OS, DB, other environment details and versions:**
MacOS Monterey

**Comment by gayaldassanayake** [comment]
The exception oriiginates from `io.ballerina.projects.SemanticVersion -> from` class due to the invalid version given here `2.3.0_temp`. The `ProjectException` thrown here is not handled in the `io.ballerina.projects.PackageVersion -> from`.

This can be fixed by replacing,
``` java
PackageVersion.from(SemanticVersion.from(versionString));
```
with,
``` java
PackageVersion version;
try {
    version = PackageVersion.from(SemanticVersion.from(versionString));
} catch (ProjectException e) {
    version = ProjectUtils.defaultVersion();
}
```

However I wasn't able to recreate the issue so far. The error has been occurred when running `org.ballerinalang.langserver.codeaction.providers.imports.PullModuleCodeAction` when pulling a package from the central. However, the `bal pack` command throws an error if the version is not in the semver format.

**Comment by malinthar** [comment]
This is due to a corrupted package(s) in the remote repository of ballerina user home(.ballerina). When we download a package from central if the download process is interrupted (internet connection failure) the corrupted pack is saved with the `temp` suffix.  When we try to load such packs from the ballerina user home using a resolution request `ILLEGAL CHAR` exception is thron. I think the Project API behavior is correct. Form the LS side in the LSPackageLoader we can explicitly catch and handle this exception and proceed with loading the remaining packages.

**Comment by gayaldassanayake** [comment]
Fixed for 2201.1.x with https://github.com/ballerina-platform/ballerina-lang/pull/37218
